python_modules/dagster/dagster/_core/definitions/auto_materialize_rule.py

```diff
@@ -629,6 +645,85 @@ def unpack(
         check.failed(f"Unexpected class {self.klass}")
 
 
+@whitelist_for_serdes
+class SkipOnNotAllParentsUpdatedRule(
+    AutoMaterializeRule,
+    NamedTuple(
+        "_SkipOnNotAllParentsUpdatedRule", [("require_update_on_all_parent_partitions", bool)]
+    ),
+):
+    """An auto-materialize rule that enforces that an asset can only be requested for execution
+    if all parents have been materialized since its last materialization.
+
+    Properties:
+        require_update_on_all_parent_partitions (Optional[bool]): If true, requires all upstream
```

Comment:
I would name use "for" or "to" instead of "on".  We use "on" elsewhere in situations when we mean "do X _on_ Y happening"